### ISSUE TITLE
Client libraries should reject '*' when subscribing for push.

### ISSUE THREAD

**Issue by tcard** [ably/docs#343]
It should be rejected as a value of DeviceDetails.clientId and PushChannelSubscription.clientId, and consequently it should be rejected when calling PushChannel.subscribeClient.

**Comment by mattheworiordan** [comment]
@paddybyers @funkyboy are you aware of this issue?  Should this be included in the 1.1 spec, and is this addressed in any libraries?

**Comment by paddybyers** [comment]
Yes, and from memory I addressed it in the `device-token` realtime push changes, and it is addressed in the realtime tests. The results of that work need to be reflected in the library docs, so we should check this point when we get to do that
